### PR TITLE
DCS-863 Tidying up API/data layer

### PR DESCRIPTION
**Pull request by andrewrlee**
Renaming references to elite2 to prison api. 
Creating a `/data/index.ts`. 
Extracting out generic API builders

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,10 +23,7 @@ The app requires:
 To start the main services excluding the use of force app: 
 `docker-compose up --scale=use-of-force=0`
 
-To start all services including use of force and new-nomis: 
-`docker-compose -f docker-compose.yml -f docker-compose.newnomis.yml up`
```
Comment (andrewrlee) [inline README.md:27]: this was quite out of date and not super useful so rather than fix I just got rid.

File: docker-compose.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ services:
       test: ["CMD", "curl", "-f", "http://localhost:8080/auth/health"]
     environment:
       SPRING_PROFILES_ACTIVE: dev,token-verification
-      TOKENVERIFICATION_ENDPOINT_URL: "http://token-verification-api:8080"
+      TOKENVERIFICATION_API_URL: "http://token-verification-api:8080"
```
Comment (andrewrlee) [inline docker-compose.yml:15]: moving to usual convention for env var

File: feature.env [hunk 3]
```diff
@@ -1,8 +1,8 @@
 NOTIFY_ENABLED=false
 PORT=3007
 NOMIS_AUTH_URL=http://localhost:9091/auth
-ELITE2API_ENDPOINT_URL=http://localhost:9091
-PRISONER_SEARCH_ENDPOINT_URL=http://localhost:9091/search
+PRISON_API_URL=http://localhost:9091
+PRISONER_SEARCH_API_URL=http://localhost:9091/search
```
Comment (andrewrlee) [inline feature.env:5]: moving to more sensible convention

File: server/services/drafts/draftInvolvedStaffService.test.ts [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
+import { DraftReportClient, PrisonClient, AuthClient } from '../../data'
 import { FoundUserResult } from '../../types/uof'
-import UserService from '../userService'
+import { UserService } from '..'
 import { DraftInvolvedStaffService } from './draftInvolvedStaffService'
 
-jest.mock('../../data/draftReportClient')
-jest.mock('../userService')
-jest.mock('../../data/elite2ClientBuilder')
-jest.mock('../../data/authClientBuilder')
```
Comment (andrewrlee) [inline server/services/drafts/draftInvolvedStaffService.test.ts:11]: loads nicer now being able to mock out an entire "layer" at a time

File: server/data/restClient.ts [hunk 5]
```diff
@@ -9,108 +9,125 @@ interface GetRequest {
...
+          if (err) logger.info(`Retry handler found API error with ${err.code} ${err.message}`)
+          return undefined // retry handler only for logging retries, not to influence retry logic
+        })
+        .query(query)
+        .auth(this.token, { type: 'bearer' })
+        .set(headers)
+        .responseType(responseType)
+        .timeout(this.timeoutSpec)
 
-        return raw ? result : result.body
```
Comment (andrewrlee) [inline server/data/restClient.ts:76]: this raw flag wasn't being used so stripped it out

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**